cmd/buildah/unshare.go

```diff
@@ -22,15 +26,70 @@ var (
   buildah unshare cat /proc/self/uid_map,
   buildah unshare buildah-script.sh`,
 	}
+	unshareMounts []string
 )
 
 func init() {
 	unshareCommand.SetUsageTemplate(UsageTemplate())
 	flags := unshareCommand.Flags()
 	flags.SetInterspersed(false)
+	flags.StringSliceVar(&unshareMounts, "mount", []string{}, "mount the specified containers (default [])")
 	rootCmd.AddCommand(unshareCommand)
 }
 
+func unshareMount(c *cobra.Command, mounts []string) ([]string, func(), error) {
+	var store storage.Store
+	var mountedContainers, env []string
+	if len(mounts) == 0 {
+		return nil, nil, nil
+	}
+	unmount := func() {
+		for _, mounted := range mountedContainers {
+			builder, err := openBuilder(getContext(), store, mounted)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, errors.Wrapf(err, "error loading information about build container %q", mounted))
+				continue
+			}
+			err = builder.Unmount()
+			if err != nil {
+				fmt.Fprintln(os.Stderr, errors.Wrapf(err, "error unmounting build container %q", mounted))
+				continue
+			}
+		}
+	}
+	store, err := getStore(c)
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, mountSpec := range mounts {
+		mount := strings.SplitN(mountSpec, "=", 2)
+		container := mountSpec
+		envVar := container
+		if len(mount) == 2 {
+			envVar = mount[0]
+			container = mount[1]
+		}
+		builder, err := openBuilder(getContext(), store, container)
+		if err != nil {
+			defer unmount()
```

Comment:
Why defer?  If you are going defer, should we just do a 
```
defer func() {
         if Err != nil {
             unmount()
         }
}
```

Reply:
That's a good point.  Changing it.